src/components/Modal.js

```diff
@@ -0,0 +1,72 @@
+import { Dialog, LabeledButton, useElementShouldClose } from '../';
+
+import { useRef } from 'preact/hooks';
+
+/**
+ * @typedef {import('./Dialog').DialogProps} ModalProps
```

Comment:
`Modal` takes the same props as `Dialog`. I don't know if we have a convention about renaming like I've done here (or _not_ renaming).